### ISSUE TITLE
Allow to change severity of the rules

### ISSUE THREAD

**Issue by stambolievv** [eslint-stylistic/eslint-stylistic#733]
### Configure Stylistic Rules as Warnings 

I would like to configure stylistic rules as **warnings** so they appear as non-intrusive yellow squiggles that I can ignore while developing and address later before committing.  

Right now, I often lose time fixing minor formatting issues instead of focusing on actual functionality because **real 
 breaking errors** are mixed with red stylistic violations. Since my codebase **does not allow warnings at commit time**, that is the only stage where strict enforcement matters.  

Seeing **errors in IDE** is distracting because errors typically indicate:  
- The code will not compile.  
- There is a bug that prevents it from working.  

Ideally, the linter should allow all stylistic rules to be treated as warnings. Otherwise, I am forced to **explicitly redefine every recommended rule**, which is unnecessary overhead. 

### Suggested solution

We can modify the existing structure using the `customize` method by introducing a new option:  

```js
const {
  ...
  semi = false,
  severity = "error",
} = options;
```

This allows us to dynamically adjust rule severity and apply it consistently across all stylistic rules:  

```js
...
"@stylistic/semi": [severity, semi ? "always" : "never"],
"@stylistic/semi-spacing": [severity, { after: true, before: false }],
...
```

### Validations

- [x] Follow our [Code of Conduct](https://github.com/eslint-stylistic/eslint-stylistic/blob/main/CODE_OF_CONDUCT.md)
- [x] Read the [Contributing Guide](https://eslint.style/contribute/guide).
- [x] Check that there isn't already an issue that request the same feature to avoid creating a duplicate.

### Contributes

- [x] If this feature request is accepted, I am willing to submit a PR to fix this issue

**Comment by 9romise** [comment]
Based on your description, it looks like you'd like to silence the stylistic rules.
Maybe you can take a look at this: https://github.com/antfu/eslint-config#ide-support-auto-fix-on-save

Also, it might make sense to introduce a new option to `customize`.
pr welcome :)
